File: c/leetcode/665_non_decreasing_array.c
bool checkPossibility(int* nums, int numsSize) {
    int index = -1;
    for (int i = 0; i < numsSize - 1; i++ ){
        if (nums[i] > nums[i+1]) {
            if (-1 != index) {
                return false;
            }
            index = i;
        }
    }
    
    return -1 == index || 0 == index || numsSize - 2 == index || nums[index-1] <= nums[index+1] || nums[index] <= nums[index+2];
}

File: python/leetcode/665_non_decreasing_array.py
class Solution(object):
    def checkPossibility(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        index = None
        for i in range(len(nums)-1):
            if nums[i] > nums[i+1]:
                if index != None:
                    return False
                index = i
                
        if index is None or index == 0 or index == len(nums) -2 :
            return True
        return nums[index] <= nums[index+2] or nums[index -1] <= nums[index + 1]